x/nexus/keeper/transfer.go

```diff
@@ -60,8 +67,60 @@ func (k Keeper) getTransferFee(ctx sdk.Context) (fee exported.TransferFee) {
 	return fee
 }
 
+// computeChainSurcharge computes the max surcharge fee over the baseFee
+// max_surcharge = chain.max_fee - chain.min_fee
+// surcharge = min(max_surcharge, chain.fee_rate * (amount - baseFee))
+func (k Keeper) computeChainSurcharge(ctx sdk.Context, feeInfo exported.FeeInfo, baseFee sdk.Int, amount sdk.Int) sdk.Int {
```

Comment:
Does it make sense to have this function associated to the keeper? Why not have it as a simple helper function?

Reply:
Yep, I needed it in a previous commit, but not anymore after a refactor, good catch!